test/unit/host_test.rb

```diff
@@ -710,8 +710,6 @@ def teardown
       hostgroup = FactoryGirl.create(:hostgroup, :with_puppet_orchestration)
       h = FactoryGirl.create(:host, :managed, :with_environment, :hostgroup => hostgroup)
       Setting[:manage_puppetca] = true
-      assert h.puppet_proxy.present?
-      assert h.puppetca?
```

Comment:
Actually this behavior (the two removed lines) broke discovery. I need to test this patch with discovery turned on. @ShimShtein please ping me once you rebase this, I need to test this. The problem with discovery is that when provisioning new discovered host, `new_record?` is actually false. Therefore the following two statements were never executed:

```
    assign_hostgroup_attributes(["puppet_proxy_id"]) if new_record? || (!new_record? && !puppet_proxy_id.blank?)
    assign_hostgroup_attributes(["puppet_ca_proxy_id"]) if new_record? || (!new_record? && !puppet_ca_proxy_id.blank?)
```